rdwatch/core/views/site_observation.py

```diff
@@ -215,6 +216,14 @@ class GenerateImagesSchema(Schema):
     scaleNum: None | list[int] = None
     bboxScale: None | float = 1.2
 
+    @validator('worldviewSource')
+    def validate_worldview_source(cls, v: str | None, values: dict[str, Any]):
+        if 'WV' in values['constellation'] and v is None:
+            raise ValidationError('worldviewSource is required for WV constellation')
+        elif 'WV' not in values['constellation'] and v is not None:
+            raise ValidationError('worldviewSource is only for WV constellation')
+        return v
+
```

Comment:
I don't think this validator is working properly.  I'm able to provide a None value for the worldviewSource and it doesn't error out.  It attempts to run it and the task fails because it uses the base search (PL, L8, S2) for the satellite images when in WV mode.

Reply:
There was two things happening here -
1) This validator wasn't being called at all.
2) If it did get called, accessing `values['constellation'] ` wouldn't have worked because it wouldn't have been populated yet from what i understand reading the pydantic docs. Instead, we need `root_validator` to access multiple fields in one validator.

I'm still not sure why 1) happened, but switching to `root_validator` seems to have fixed this :shrug: https://github.com/ResonantGeoData/RD-WATCH/pull/472/commits/0a9d3489b250e9af5311d407e5b7c9e9cde88c59